src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.cs

```diff
@@ -340,12 +343,13 @@ public SolutionCompilationState AddProjects(ArrayBuilder<ProjectInfo> projectInf
             modifyNewTrackerInfo: static (_, _) => { }, argModifyNewTrackerInfo: default(VoidResult),
             skipEmptyCallback: true);
 
+        // Add the new projects to the source generator execution version map.  Note: it's ok for us to have entries for
+        // non-C#/VB projects.  These will have no effect in-proc as we won't have compilation-trackers for these
+        // projects.  And, when communicating with the OOP process, we'll filter out these projects before sending them
+        // across in SolutionCompilationState.GetFilteredSourceGenerationExecutionMap.
         var versionMapBuilder = _sourceGeneratorExecutionVersionMap.Map.ToBuilder();
         foreach (var projectInfo in projectInfos)
-        {
-            if (RemoteSupportedLanguages.IsSupported(projectInfo.Language))
```

Comment:
note: my original approach was to have the crashing code filter out these project types there as well.  But i found that i had to update a whole bunch of locations (you can see that in commit ca92b8f3e8c5bae98c01bc0384f28385c969d722).  I didn't like how that was bleeding out.  So i just made the change here.  

The change we made in https://github.com/dotnet/roslyn/pull/73688 ensures that no matter what the full map is in VS, we always sync just the C#/VB portion over properly when talking to OOP.